.github/workflows/ci_pr.yml

```diff
@@ -0,0 +1,140 @@
+        sudo ln -sf /usr/bin/python2.6 /usr/bin/python
+        PATH="/usr/bin/${PATH:+:${PATH}}"
+        echo $PATH
+        echo "Installed python2.6"
+        python -V
+        curl https://bootstrap.pypa.io/2.6/get-pip.py -o get-pip.py
+        sudo env "PATH=$PATH" python get-pip.py
+
+    - uses: actions/checkout@v2
+
+    - name: Install dependencies
+      run: |
+        sudo env "PATH=$PATH" pip install -r requirements.txt
+        sudo env "PATH=$PATH" pip install -r test-requirements.txt
+    
+    - name: Test with nosetests
+      run: |
+        ./ci/nosetests.sh || EXIT_CODE=$?
+        exit $EXIT_CODE
+
+  build:
+    
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+
+          - python-version: 2.7
+            PYLINTOPTS: "--rcfile=ci/2.7.pylintrc"
+            NOSEOPTS: "--verbose --with-timer"
+            PYLINTFILES: "azurelinuxagent setup.py makepkg.py tests"
+
+          - python-version: 3.4
+            PYLINTOPTS: "--rcfile=ci/2.7.pylintrc"
+            NOSEOPTS: "--verbose --with-timer"
+            PYLINTFILES: "azurelinuxagent setup.py makepkg.py tests"
+
+          - python-version: 3.6
+            PYLINTOPTS: "--rcfile=ci/3.6.pylintrc"
+            NOSEOPTS: "--verbose --with-timer"
+            PYLINTFILES: "azurelinuxagent setup.py makepkg.py tests"
+
+          - python-version: 3.7
+            PYLINTOPTS: "--rcfile=ci/3.6.pylintrc"
+            NOSEOPTS: "--verbose --with-timer"
+            PYLINTFILES: "azurelinuxagent setup.py makepkg.py tests"
+
+          - python-version: 3.8
+            PYLINTOPTS: "--rcfile=ci/3.6.pylintrc"
+            NOSEOPTS: "--verbose --with-coverage --with-xunit --xunit-file=nosetests.__uuid__.xml
+              --with-timer --cover-erase --cover-inclusive --cover-branches --cover-package=azurelinuxagent"
+            PYLINTFILES: "azurelinuxagent setup.py makepkg.py tests"
+    
+    name: "Python ${{ matrix.python-version }} Unit Tests"
+    runs-on: ubuntu-18.04
+
+    env:
+      PYLINTOPTS: ${{ matrix.PYLINTOPTS }}
+      PYLINTFILES: ${{ matrix.PYLINTFILES }}
+      NOSEOPTS: ${{ matrix.NOSEOPTS }}
+      PYTHON_VERSION: ${{ matrix.python-version }}
+
+    steps:
+    
+    - name: Checkout WALinuxAgent repo
+      uses: actions/checkout@v2
+
+    - name: Setup Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+    
+    - name: Install dependencies
+      run: |
+        sudo env "PATH=$PATH" python -m pip install --upgrade pip
+        sudo env "PATH=$PATH" pip install -r requirements.txt
+        sudo env "PATH=$PATH" pip install -r test-requirements.txt
+
+    - name: Run pylint
+      run: |
+        pylint $PYLINTOPTS --jobs=0 $PYLINTFILES
+
+    - name: Test with nosetests
+      if: always()
+      run: |
+        ./ci/nosetests.sh || EXIT_CODE=$?
+        exit $EXIT_CODE
+
+    - name: Run Coverage
```

Comment:
How to read and parse this data? Like from the `checks` page it wasnt very clear how to read the data. 
Is there a way we can add it to the PR page (similar to how we had it for travis)?

Reply:
It generates a codecov report.
Also will post the test results at the end of the checks page.

Reply:
I saw a link and clicked on it but it showed nothing. Can you post the link just incase I picked the wrong one?

>but we should add another task for publish unit tests results

That would be helpful yes :)

Reply:
This is the link I tried - 

> -> View reports at https://codecov.io/github/Azure/WALinuxAgent/commit/39e5f5e2e3ccd2294044a705624f245ec6240204

Reply:
Just scroll down a bit, you will see
![image](https://user-images.githubusercontent.com/5965677/101524849-54997c80-393f-11eb-8e33-658e0db11b84.png)

Then click on details
ex : https://github.com/Azure/WALinuxAgent/pull/2097/checks?check_run_id=1518915394

Reply:
But these are details for the whole run. I meant specifically for codecov (the link I pasted above is from the details of Pyton 3.8 Unit Tests)

Reply:
there was no code change here, so codecov didnt show a diff I think

Reply:
Hmm interesting, @kevinclark19a can you give this a test run? Like change some random code and ensure the codecov is being hit? Else might as well remove this check (because we dont have it now either)

Reply:
If you connect with your github account, and go in files, you can see the test coverage
By default, codecov opens on the diff page.

Reply:
nvm the codecov report showed up finally (it probably takes some time to reflect in the portal)
 
![image](https://user-images.githubusercontent.com/41490930/101527796-52392180-3943-11eb-8caf-db4a12e4d73f.png)

Reply:
I think it was waiting for Travis to run (it appeared at the same time Travis finished) that's weird

Reply:
hmm weird, we're gonna remove travis anyways so even if this is a thing, its a fleeting pain lol